src/problem.jl

```diff
@@ -1,13 +1,15 @@
 abstract type AbstractOptimizationProblem end
 
-struct OptimizationProblem{F,X,P,B,K} <: AbstractOptimizationProblem
+struct OptimizationProblem{F,X,P,B,LC,UC,K} <: AbstractOptimizationProblem
     f::F
     x::X
     p::P
     lb::B
     ub::B
+    lcons::LC
+    ucons::UC
     kwargs::K
-    function OptimizationProblem(f, x; p=DiffEqBase.NullParameters(), lb = nothing, ub = nothing, kwargs...)
-        new{typeof(f), typeof(x), typeof(p), typeof(lb), typeof(kwargs)}(f, x, p, lb, ub, kwargs)
+    function OptimizationProblem(f, x; p=DiffEqBase.NullParameters(), lb = [], ub = [], lcons = [], ucons = [], kwargs...)
```

Comment:
why instead of nothing?

Reply:
Because when passing to `TwiceDifferentiableConstraint` constructor we need empty array in case of no bounds, https://github.com/JuliaNLSolvers/NLSolversBase.jl/blob/master/src/objective_types/constraints.jl#L174

Reply:
But that can be swapped in the function call.